components/compositing/compositor.rs

```diff
@@ -1422,6 +1435,40 @@ impl<Window: WindowMethods> IOCompositor<Window> {
             self.webrender.render(self.frame_size);
         });
 
+        // If there are pending paint metrics, we check if any of the painted epochs is
+        // one of the ones that the paint metrics recorder is expecting . In that case,
+        // we get the current time, inform the layout thread about it and remove the
+        // pending metric from the list.
+        if !self.pending_paint_metrics.is_empty() {
+            let paint_time = precise_time_ns() as f64;
+            let mut to_remove = Vec::new();
+            // For each pipeline
+            for (id, _) in &self.pipeline_details {
+                // if its layout thread is expecting the paint time for a specifc frame
+                if let Some(pending_epoch) = self.pending_paint_metrics.get(id) {
```

Comment:
Why not iterate over just pending metrics tables keys instead of all the pipelines?